api/src/main/java/org/pmiops/workbench/jira/JiraServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package org.pmiops.workbench.jira;
+
+import org.json.JSONObject;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.google.CloudStorageService;
+import org.pmiops.workbench.jira.api.JiraApi;
+import org.pmiops.workbench.jira.model.FieldsDetails;
+import org.pmiops.workbench.jira.model.IssueRequest;
+import org.pmiops.workbench.jira.model.IssueResponse;
+import org.pmiops.workbench.jira.model.IssueType;
+import org.pmiops.workbench.jira.model.ProjectDetails;
+import org.pmiops.workbench.model.BugReport;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import javax.inject.Provider;
+
+@Service
+public class JiraServiceImpl implements JiraService {
+
+  private static final Logger log = Logger.getLogger(JiraServiceImpl.class.getName());
+  private JiraApi api = new JiraApi();
+  private final Provider<WorkbenchConfig> configProvider;
+  private CloudStorageService cloudStorageService;
+
+  @Autowired
+  public JiraServiceImpl(Provider<WorkbenchConfig> configProvider,
+      Provider<CloudStorageService> cloudStorageServiceProvider) {
+    this.configProvider = configProvider;
+    this.cloudStorageService = cloudStorageServiceProvider.get();
+  }
+
+  /**
+   * Sets username password for JIRA api
+   */
+  private void setJiraCredentails(){
+    JSONObject jiraCredentails = cloudStorageService.getJiraCredentials();
+    api.getApiClient().setUsername(jiraCredentails.getString("username"));
+    api.getApiClient().setPassword(jiraCredentails.getString("password"));
+  }
+
+
+  @Override
+  public String createIssue(BugReport bugReport) throws ApiException {
+    setJiraCredentails();
+    IssueRequest issueDetails = new IssueRequest();
+    IssueType issueType = new IssueType();
+    FieldsDetails fieldsDetail = new FieldsDetails();
+    ProjectDetails projectDetails = new ProjectDetails();
+
+    projectDetails.setKey(configProvider.get().jira.projectKey);
+
+    fieldsDetail.setDescription(String.format("%s %nContact Email: %s",bugReport.getReproSteps(),
+        bugReport.getContactEmail()));
+    fieldsDetail.setSummary(bugReport.getShortDescription());
+    fieldsDetail.setProject(projectDetails);
+
+    issueType.setName(IssueTypeEnum.Bug.name());
+    fieldsDetail.setIssuetype(issueType);
+
+    issueDetails.setFields(fieldsDetail);
+    api.getApiClient().setDebugging(true);
+    IssueResponse response = api.createIssue(issueDetails);
+    return response.getKey();
+  }
+
+
+  @Override
+  public void uploadAttachment(String issueKey, String attachmentName, Object content) throws ApiException {
```

Comment:
I know you're thinking of other ways to pass the attachment content. 
One thought is to have different, overloadable methods that deal with the specific content types we support. So, for example, we could only support String content right now and in the future, support other types.

Reply:
Looking more closely at the content that is returned by the jupyter api, the results are of type `JupyterContents`, from which, we get the logContent string value. It might make sense to support uploading the full object itself instead of looking at the string content.

Reply:
I think `content` should definitely be a `String` at this point, where the string is the file contents of the attachment to upload. The goal with this attachment was to basically push the exact logfile as we saw it on the Jupyter server onto the Jira ticket.

The tempfile stuff is a bit messy, but I'm not sure I see a clear way around with the Swagger java codegen (I looked through the generated source). Maybe this approach will suffice for now - we'd also discussed just pushing these logs into GCS for access control purposes, so that would be another approach for bypassing this problem (and you're more than welcome to do that in this PR, if that makes life easier for you somehow - I'm thinking it probably won't).